### PR TITLE
Enable restoration of Byron Sequential wallet from xpub

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

ADP-334

# Overview

- 3ef0e95b9efe186e3c4e53dce7b2df206cf192da
  :round_pushpin: **extend swagger specification to add xpub restoration for byron sequential wallets**
  
- 5deb64c4b24da8f7faa76f373a074c491e298c9e
  :round_pushpin: **extend swagger specification to add xpub restoration for byron sequential wallets**
  
- 7cc16c15d81ac1a6e793e25416666ace05a45ced
  :round_pushpin: **quick-n-dirty hack of the documentation to explicitely name oneof labels**
  Instead of mere 'object', they now have significant names

- 69257ad1e486d3867c7d73b7deece431a58e3982
  :round_pushpin: **show 'standard' wallet as default in API documentation for shelley**
  Instead of showing the import from account public key as default

- 58ab3221e979afa2cfa19a528a7ee742ce622955
  :round_pushpin: **extend Byron servant API to allow restoring Byrons sequential wallet from xpub**
  
- bd8c008bfa0432884a9266a6e114b0fb906804cf
  :round_pushpin: **generalize a bit some DSL function to work with both Byron & Shelley wallets**
  
- d69de747da9bc734d1bf1790650e1be677594ec8
  :round_pushpin: **fix JSON parsers for posting Byron wallets: there's a new case**
  
- 0ef9a3181c754423075ab923f8cc74e5909d1242
  :round_pushpin: **wire hardware wallet integration tests with cardano-node-byron**
  
- 773ad3a51256d6459bda2b725feef07cb0e1ee89
  :round_pushpin: **remove hack in favor and use schema 'title' property instead**
  
- c3b2273ecbd4b14a5015d68f9a42134fdcb0baf9
  :round_pushpin: **extend API specification to support coin selection for byron wallets**
  
- 08bfe99b67c52d1eacc59b99dfeb6f813e1613fb
  :round_pushpin: **turn on coin selection endpoint for byron wallets in byron & shelley servers**

# Comments

Screenshot from 2020-06-03 14-52-27

Screenshot from 2020-06-03 14-52-36

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Api/Server.hs [hunk 1]
```diff
@@ -562,15 +564,16 @@ postAccountWallet
...
-    -> Handler ApiWallet
-postAccountWallet ctx liftKey body = do
+    -> Handler w
+postAccountWallet ctx mkWallet liftKey body = do
     let state = mkSeqStateFromAccountXPub (liftKey accXPub) g
     void $ liftHandler $ initWorker @_ @s @k ctx wid
         (\wrk -> W.createWallet  @(WorkerCtx ctx) @s @k wrk wid wName state)
         (\wrk -> W.restoreWallet @(WorkerCtx ctx) @s @t @k wrk wid)
-    fst <$> getWallet ctx (mkShelleyWallet @_ @s @t @k) (ApiT wid)
+    fst <$> getWallet ctx mkWallet (ApiT wid)
```
Comment (paweljakubas) [inline lib/core/src/Cardano/Wallet/Api/Server.hs:577]: :+1:

**Review by paweljakubas — APPROVED** [review]
looks good. would be good to have integration test illustrating the case. Something like we create wallet like before this PR, then do some txs to it, then delete it, and finally restore this way and checking the balance

File: specifications/api/index.html [hunk 2]
```diff
@@ -10,10 +10,55 @@
...
 <body>
   <redoc spec-url="./swagger.yaml"></redoc>
   <script src="https://cdn.jsdelivr.net/npm/redoc@2.0.0-rc.23/bundles/redoc.standalone.js" async defer></script>
+  <script>
+  /* Little hack to be able to 'name' the oneof selector when showing multiple body choices.
+   * By default, redoc shows a little 'object' labels, which is not really visible nor helpful.
+   *
+   * TODO: Make this configurable upstream in ReDoc.
+   */
+  window.onload = function () {
```
Comment (paweljakubas) [inline specifications/api/index.html]: :1st_place_medal:

**Comment by KtorZ** [comment]
bors try

**Comment by KtorZ** [comment]
bors try

**Comment by darko-mijic** [comment]
❤️

**Comment by KtorZ** [comment]
WHAT IS YOUR PROBLEM HYDRA FFS

**Comment by disassembler** [comment]
bors try

**Comment by KtorZ** [comment]
@disassembler there's also a failure in the test suite that needs resolution first.

**Comment by KtorZ** [comment]
bors try

**Pull request merged**